### PR TITLE
Simplify a couple of methods

### PR DESCRIPTION
**Pull request by pavlik4k**

### CODE REVIEW

File: lib/http/response/parser.rb [hunk 1]
```diff
@@ -34,17 +34,12 @@ def on_headers_complete(headers)
...
-        end
+        @chunk ||= ''
+        @chunk << chunk
       end
 
       def chunk
-        chunk  = @chunk
-        @chunk = nil
-        chunk
+        remove_instance_variable(:@chunk)
```
Comment (sferik) [inline lib/http/response/parser.rb]: missing: `if defined?(:@chunk)`

**Comment by tarcieri** [comment]
This is not semantically equivalent (and breaks the tests), nor would I consider it a simplification

**Comment by pavlik4k** [comment]
Actually it doesn't break specs. You should check your ci

**Comment by tarcieri** [comment]
On Friday, April 29, 2016, pavlik4k <notifications@github.com> wrote:

> Actually it doesn't break specs. You should check your ci
>
No, your code is broken (see @sferik's earlier comment):

 Failure/Error: remove_instance_variable(:@chunk)

 NameError:

       instance variable @chunk not defined



-- 
Tony Arcieri

File: lib/http/response/parser.rb [hunk 2]
```diff
@@ -34,17 +34,12 @@ def on_headers_complete(headers)
       end
 
       def on_body(chunk)
-        if @chunk
-          @chunk << chunk
-        else
-          @chunk = chunk
-        end
+        @chunk ||= ''
```
Comment (ixti) [inline lib/http/response/parser.rb]: this allocates new string.
and honestly i don't see significant improvement here. if/else condition looks easier to read imho in this particular case.

**Comment by ixti** [comment]
I don't find this to be improvement either.

File: lib/http/response/parser.rb (hunk 2 above)
Comment (tarcieri) [inline lib/http/response/parser.rb]: Yeah +1, this adds an additional allocation and makes the code more complicated, in addition to just breaking things.

**Pull request closed without merging**